distribution_metadata.c

```diff
@@ -450,7 +452,15 @@ ColumnNameToColumn(Oid relationId, char *columnName)
 	const Index tableId = 1;
 	const Index columnLevelsUp = 0;
 
-	AttrNumber columnId = get_attnum(relationId, columnName);
+	/* check length to prevent failed Assert for long names in get_attnum */
+	if (strnlen(columnName, NAMEDATALEN) == NAMEDATALEN)
+	{
+		ereport(ERROR, (errcode(ERRCODE_NAME_TOO_LONG), errmsg("columnName too long"),
```

Comment:
This made me double-check other error messages for variable names. I found a few places, all of them [in this region](https://github.com/citusdata/pg_shard/blob/d1a335dae59413b2db47896d69b4dab7ac1b1224/create_shards.c#L188-L222).

In that case, the messages are actually referring to user-specified function arguments, so I think that's an argument to actually change them to match the underscore-separated variable names used in the SQL. So I'll do that.

For here, though… this is an internal C method and the error is about calling with bad C parameters. So it's more of an internal error. That would be a case to make this an `Assert`, but this could be triggered by someone simply updating values in the partition table using SQL.

That might be a case to redefine that column as the `name` type rather than `text`, but in lieu of that I'll check other PostgreSQL code to see whether any internal functions complain using their own parameter names even though the end-user might not even know what those variables are talking about.

Otherwise, I might do away with the _column_ word altogether and see what kind of errors I can dig up that reference names being too long.